lib/developer_portal/app/controllers/developer_portal/admin/messages/outbox_controller.rb

```diff
@@ -35,15 +35,19 @@ def destroy
 
   def create
     @message.enqueue! :to => current_account.provider_account.id
-    @notice = 'Message was sent.'
 
-    respond_to do |format|
-      format.html do
-        flash[:notice] = @notice
-        redirect_to admin_messages_root_path
+    if @message.body.present? && @message.subject.present?
+      @notice = 'Message was sent.'
+      respond_to do |format|
+        format.html do
+          flash[:notice] = @notice
+          redirect_to admin_messages_root_path
+        end
+        format.js
       end
-
-      format.js
+    else
+      flash[:error] = 'Please fill subject and body.'
```

Comment:
We should consider that maybe it could be just subject or body what the user is missing, right?

Reply:
Yes, but I meant that what could be missing could be just subject or body, right? But I get the point, I was just wondering.

Reply:
I covered this in test case with below mentioned case hope that helps